core/src/main/java/org/elasticsearch/cluster/metadata/IndexMetaData.java

```diff
@@ -812,6 +827,15 @@ public int numberOfReplicas() {
             return settings.getAsInt(SETTING_NUMBER_OF_REPLICAS, -1);
         }
 
+        public Builder routingPartitionSize(int routingPartitionSize) {
+            settings = Settings.builder().put(settings).put(SETTING_ROUTING_PARTITION_SIZE, routingPartitionSize).build();
+            return this;
+        }
+
+        public int routingPartitionSize() {
+            return settings.getAsInt(SETTING_ROUTING_PARTITION_SIZE, -1);
```

Comment:
We could but I was going for consistency with the lines above. For example, the ``numberOfReplicas()`` and ``numberOfShards()`` in the builder each return a value of -1 when they are not set as opposed to the default for the setting.

I got the impression that we would want to return -1 here to let callers differentiate between the default value and an unset value.